spanner/transaction.go

```diff
@@ -1056,6 +1076,10 @@ func (t *ReadWriteTransaction) batchUpdateWithOptions(ctx context.Context, stmts
 		return nil, ToSpannerError(err)
 	}
 
+	if t.tx == nil && resp.ResultSets != nil && len(resp.ResultSets) > 0 && resp.ResultSets[0].Metadata != nil {
```

Comment:
Shouldn't this also include a `&& resp.ResultSets[0].Metadata.Transaction.GetId() != nil` (or similar)? I think this will clear the current transaction ID if you execute two batch DML operations in one transaction.

Reply:
Updated to call this only when TransactionSelector::Begin is set. we are making sure that only one req will be executed with TransactionSelector::Begin in a read/write transaction.